activerecord/lib/active_record/validations.rb

```diff
@@ -82,6 +82,21 @@ def perform_validations(options = {}) # :nodoc:
       options[:validate] == false || valid?(options[:context])
     end
   end
+
+  module Persistence
```

Comment:
Yes, many ways! In an [earlier](https://github.com/rails/rails/pull/27281/commits/3a7562f2c47c8d36a1f69ae2c77ff872f3320ced) iteration in fact this lived in Persistence directly. @sgrif and I discussed moving it but I'm not in love with it here either.